src/t_string.c

```diff
@@ -99,7 +99,8 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
         return;
     }
 
-    setkey_flags |= (flags & OBJ_KEEPTTL) ? SETKEY_KEEPTTL : 0;
+    /* When expire is not NULL, we avoid deleting so it can be updated later instead of first deleted and then created again. */
```

Comment:
```suggestion
    /* When expire is not NULL, we avoid deleting the TTL so it can be updated later instead of being deleted and then created again. */
```